tests/commands/test_build.py

```diff
@@ -0,0 +1,692 @@
+    testfile1 = pathlib.Path(str(testfile1))  # comparison below don't work well in Py3.5
+    testfile2 = pathlib.Path(str(testfile2))  # comparison below don't work well in Py3.5
+    assert resp == [testfile1, testfile2]
+
+
+def test_validator_requirement_default_present_ok(tmp_path):
+    """'requirement' param: default value when a requirements.txt is there and readable."""
+    default_requirement = tmp_path / 'requirements.txt'
+    default_requirement.touch()
+
+    validator = Validator()
+    validator.basedir = tmp_path
+    resp = validator.validate_requirement(None)
+    assert resp == [default_requirement]
+
+
+def test_validator_requirement_default_present_not_readable(tmp_path):
+    """'requirement' param: default value when a requirements.txt is there but not readable."""
+    default_requirement = tmp_path / 'requirements.txt'
+    default_requirement.touch(0o230)
+
+    validator = Validator()
+    validator.basedir = tmp_path
+    resp = validator.validate_requirement(None)
+    assert resp == []
+
+
+def test_validator_requirement_default_missing(tmp_path):
+    """'requirement' param: default value when no requirements.txt is there."""
+    validator = Validator()
+    validator.basedir = tmp_path
+    resp = validator.validate_requirement(None)
+    assert resp == []
+
+
+def test_validator_requirement_absolutized(tmp_path, monkeypatch):
+    """'requirement' param: check it's made absolute."""
+    # change dir to the temp one, where we will have the reqs file
+    testfile = tmp_path / 'reqs.txt'
+    testfile.touch()
+    monkeypatch.chdir(tmp_path)
+
+    validator = Validator()
+    resp = validator.validate_requirement(['reqs.txt'])
+    testfile = pathlib.Path(str(testfile))  # comparison below don't work well in Py3.5
+    assert resp == [testfile]
+
+
+def test_validator_requirement_expanded():
+    """'requirement' param: expands the user-home prefix."""
+    home = pathlib.Path("~").expanduser()
+    testfd, testpath = tempfile.mkstemp(dir=str(home))
+    try:
+        testpath = pathlib.Path(testpath)
+
+        validator = Validator()
+        relative_testpath = "~/{}".format(testpath.relative_to(home))
+        resp = validator.validate_requirement([relative_testpath])
+    finally:
+        testpath.unlink()
+    assert resp == [testpath]
+
+
+def test_validator_requirement_exist():
+    """'requirement' param: checks that the file exists."""
+    validator = Validator()
+    expected_msg = "the requirements file was not found: '/not_really_there.txt'"
+    with pytest.raises(CommandError, match=expected_msg):
+        validator.validate_requirement(['/not_really_there.txt'])
+
+
+# --- Polite Executor tests
+
+def test_politeexec_base(caplog):
+    """Basic execution."""
+    caplog.set_level(logging.ERROR, logger="charmcraft")
+
+    cmd = ['echo', 'HELO']
+    retcode = polite_exec(cmd)
+    assert retcode == 0
+    assert not any("Execution ended" in rec.message for rec in caplog.records)
```

Comment:
This seems like a very specific test that isn't going to fail when something gets changed. I'm not sure what a fix would be. Maybe assert that there are no records at ERROR level?

Reply:
Good idea! Changed to verify that there are no ERROR records.